git-duet-prepare-commit-msg/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/git-duet/git-duet"
+	"github.com/pborman/getopt"
+)
+
+func main() {
+
+	getopt.Parse()
+	commitMsgFile := getopt.Args()[0]
+
+	configuration, err := duet.NewConfiguration()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	var gitConfig *duet.GitConfig
+	if configuration.Global {
+		gitConfig = &duet.GitConfig{
+			Namespace:     configuration.Namespace,
+			Scope:         duet.Global,
+			SetUserConfig: configuration.SetGitUserConfig,
+		}
+	} else {
+		gitConfig, err = duet.GetAuthorConfig(configuration.Namespace, configuration.SetGitUserConfig)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	committers, err := gitConfig.GetCommitters()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	if committers == nil || len(committers) == 0 {
+		os.Exit(0)
```

Comment:
I think there is just a single path how users can opt in, namely by running `git duet` with `GIT_DUET_CO_AUTHORED_BY` set.

I think in your above example, it's okay that the commit doesn't set `js` as author since the user didn't run `git solo` with either `GIT_DUET_SET_GIT_USER_CONFIG` or `GIT_DUET_CO_AUTHORED_BY` (the latter implicitly setting `GIT_DUET_SET_GIT_USER_CONFIG`).

Reply:
Right, I'm just thinking that people might have an expectation, in my example, that `GIT_DUET_CO_AUTHORED_BY` would pick up the previous `git solo` even without `GIT_DUET_SET_USER_CONFIG`. I'm willing to concede that it feels likely that users will set that variable and then call `git duet` though.